python/tests/test_types.py

```diff
@@ -0,0 +1,163 @@
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import pytest
+
+from iceberg.types import (
+    BinaryType,
+    BooleanType,
+    DateType,
+    DecimalType,
+    DoubleType,
+    FixedType,
+    FloatType,
+    IntegerType,
+    ListType,
+    LongType,
+    MapType,
+    NestedField,
+    StringType,
+    StructType,
+    TimestampType,
+    TimestamptzType,
+    TimeType,
+    UUIDType,
+)
+
+
+@pytest.mark.parametrize(
+    "input_type",
+    [
+        BooleanType,
+        IntegerType,
+        LongType,
+        FloatType,
+        DoubleType,
+        DateType,
+        TimeType,
+        TimestampType,
+        TimestamptzType,
+        StringType,
+        UUIDType,
+        BinaryType,
+    ],
+)
+def test_repr_primitive_types(input_type):
+    assert input_type == eval(repr(input_type))
+
+
+def test_fixed_type():
+    type_var = FixedType(length=5)
+    assert type_var.length == 5
+    assert str(type_var) == "fixed[5]"
+    assert repr(type_var) == "FixedType(length=5)"
+    assert str(type_var) == str(eval(repr(type_var)))
+
+
+def test_decimal_type():
+    type_var = DecimalType(precision=9, scale=2)
+    assert type_var.precision == 9
+    assert type_var.scale == 2
+    assert str(type_var) == "decimal(9, 2)"
+    assert repr(type_var) == "DecimalType(precision=9, scale=2)"
+    assert str(type_var) == str(eval(repr(type_var)))
+
+
+def test_struct_type():
+    type_var = StructType(
+        [
+            NestedField(True, 1, "optional_field", IntegerType),
+            NestedField(False, 2, "required_field", FixedType(5)),
+            NestedField(
+                False,
+                3,
+                "required_field",
+                StructType(
+                    [
+                        NestedField(True, 4, "optional_field", DecimalType(8, 2)),
+                        NestedField(False, 5, "required_field", LongType),
+                    ]
+                ),
+            ),
+        ]
+    )
+    assert len(type_var.fields) == 3
+    assert str(type_var) == str(eval(repr(type_var)))
+
+
+def test_list_type():
+    type_var = ListType(
+        NestedField(
+            False,
+            1,
+            "required_field",
+            StructType(
+                [
+                    NestedField(True, 2, "optional_field", DecimalType(8, 2)),
+                    NestedField(False, 3, "required_field", LongType),
+                ]
+            ),
+        )
+    )
+    assert isinstance(type_var.element.type, StructType)
+    assert len(type_var.element.type.fields) == 2
+    assert type_var.element.field_id == 1
+    assert str(type_var) == str(eval(repr(type_var)))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Use of possibly insecure function - consider using safer ast.literal_eval. https://bandit.readthedocs.io/en/latest/blacklists/blacklist_calls.html#b307-eval